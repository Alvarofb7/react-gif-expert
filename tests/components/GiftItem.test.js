import { GifItem } from "../../src/components/GifItem"
import { render, screen } from "@testing-library/react"

describe('Pruebas en <GiftItem />', () => {

    const props = {
        title: "Titulo",
        url: "https://alvaro.com/image.jpg"
    };

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifItem {...props} />);

        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem {...props} />);

        // screen.debug();
        // expect(screen.getByRole("img").alt).toBe(props.title);
        // expect(screen.getByRole("img").src).toBe(props.url);

        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(props.url);
        expect(alt).toBe(props.title);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem {...props} />);
        
        expect(screen.getByText(props.title)).toBeTruthy();
    });

});