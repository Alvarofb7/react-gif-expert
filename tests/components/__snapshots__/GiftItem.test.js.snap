// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://alvaro.com/image.jpg"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
